app/renderer/components/Inspector/GestureEditor.js

```diff
@@ -153,137 +159,142 @@ class GestureEditor extends Component {
 
   // This converts all the coordinates in the gesture to px/%
   convertCoordinates (type) {
+    const {width, height} = this.props.windowSize;
     const {coordType, pointers} = this.state;
     const newPointers = JSON.parse(JSON.stringify(pointers));
     if (type !== coordType) {
-      newPointers.map((pointer) => {
-        (pointer.ticks).map((tick) => {
-          if (tick.type === POINTER_MOVE) {
+      for (const pointer of newPointers) {
+        for (const tick of pointer.ticks) {
+          if (tick.type === POINTER_TYPES.POINTER_MOVE) {
             if (type === COORD_TYPE.PIXELS) {
-              tick.x = this.percentageToPixels(tick.x, true);
-              tick.y = this.percentageToPixels(tick.y, false);
+              tick.x = percentageToPixels(tick.x, true, width, height);
+              tick.y = percentageToPixels(tick.y, false, width, height);
             } else {
-              tick.x = this.pixelsToPercentage(tick.x, true);
-              tick.y = this.pixelsToPercentage(tick.y, false);
+              tick.x = pixelsToPercentage(tick.x, true, width, height);
+              tick.y = pixelsToPercentage(tick.y, false, width, height);
             }
           }
-          return tick;
-        });
-        return pointer;
-      });
+        }
+      }
     }
     return newPointers;
   }
 
-  pixelsToPercentage (pixel, isX) {
+  getDefaultMoveDuration (ticks, tickId, x2, y2) {
     const {width, height} = this.props.windowSize;
-    if (!isNaN(pixel)) {
-      if (isX) {
-        return parseFloat(((pixel / width) * 100).toFixed(1), 10);
-      } else {
-        return parseFloat(((pixel / height) * 100).toFixed(1), 10);
+    const {coordType} = this.state;
+    const {PERCENTAGES} = COORD_TYPE;
+    const ticksExceptCurrent = ticks.filter((tick) => tick.id !== tickId);
+    const prevPointerMoves = [];
+    for (const tick of ticksExceptCurrent) {
+      if (tick.type === POINTER_MOVE && tick.x !== undefined && tick.y !== undefined) {
+        prevPointerMoves.push({x: tick.x, y: tick.y});
       }
-    } else {
-      return undefined;
     }
-  }
-
-  percentageToPixels (percentage, isX) {
-    const {width, height} = this.props.windowSize;
-    if (!isNaN(percentage)) {
-      if (isX) {
-        return Math.round(width * (percentage / 100));
-      } else {
-        return Math.round(height * (percentage / 100));
+    const len = prevPointerMoves.length;
+    if (len !== 0) {
+      const obj = {x1: prevPointerMoves[len - 1].x, y1: prevPointerMoves[len - 1].y, x2, y2};
+      if (coordType === PERCENTAGES) {
+        obj.x1 = percentageToPixels(obj.x1, true, width, height);
+        obj.y1 = percentageToPixels(obj.y1, false, width, height);
+        obj.x2 = percentageToPixels(obj.x2, true, width, height);
+        obj.y2 = percentageToPixels(obj.y2, false, width, height);
       }
+      const eq = (val1, val2) => Math.sqrt((val1 ^ 2) + (val2 ^ 2));
```

Comment:
what is `eq`? seems like a name like `distance` or `length` might be easier to understand?